__tests__/app.test.js

```diff
@@ -152,40 +152,65 @@ describe("/api/articles/:article_id", () => {
-      votes: expect.any(Number),
-      created_at: expect.any(String),
-      article_img_url: expect.any(String),
-      article_id: expect.any(Number),
-      comment_count: expect.any(Number),
-    };
-    return request(app)
-      .get("/api/articles")
-      .expect(200)
-      .then(({ body }) => {
-        const { articles } = body;
-        expect(articles.length).toBe(13);
-        articles.forEach((article) => {
-          expect(article).toMatchObject(exampleArticle);
+  describe("GET /api/articles", () => {
+    test("GET 200: Responds with array of all article objects.", () => {
+      const exampleArticle = {
+        title: expect.any(String),
+        topic: expect.any(String),
+        author: expect.any(String),
+        votes: expect.any(Number),
+        created_at: expect.any(String),
+        article_img_url: expect.any(String),
+        article_id: expect.any(Number),
+        comment_count: expect.any(Number),
+      };
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          expect(articles.length).toBe(13);
+          articles.forEach((article) => {
+            expect(article).toMatchObject(exampleArticle);
+          });
         });
-      });
+    });
+
+    test("GET 200: Responds with all article objects sorted by date in descending order", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          expect(articles).toBeSortedBy("created_at", { descending: true });
+          articles.forEach((article) => {
+            expect(article).not.toHaveProperty("body");
+          });
+        });
+    });
   });
 
-  test("GET 200: Responds with all article objects sorted by date in descending order", () => {
-    return request(app)
-      .get("/api/articles")
-      .expect(200)
-      .then(({ body }) => {
-        const { articles } = body;
-        expect(articles).toBeSortedBy("created_at", { descending: true });
-        articles.forEach((article) => {
-          expect(article).not.toHaveProperty("body");
+  describe("GET /api/articles (topic query)", () => {
+    test("GET 200: Responds with array of all articles for given topic value.", () => {
+      return request(app)
+        .get("/api/articles?topic=cats")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          expect(articles.length).toBe(1);
+          articles.forEach((article) => {
+            expect(article.topic).toBe("cats");
+          });
         });
-      });
+    });
+    test("GET 404: Responds with error message when passed a topic value that doesnt exist.", () => {
+      return request(app)
+        .get("/api/articles?topic=weather")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body).toEqual({ message: "Topic not found." });
+        });
+    });
```

Comment:
Nice error handler here, it would be useful to also think about one more possibility: what if the topic is valid and does exist in the database but there happens to be no articles associated with it?